### DISCUSSION TITLE
Apply show rule only inside math mode

### DISCUSSION THREAD

**Discussion by NicoWeio** [typst/typst#2273] [Q&A] [answered]
I'm currently using `#show "L": math.cal`, but this (obviously) applies to all L's, not just those in equations. 
I know I could do something like `#let LL = $cal(L)$`, but a) I'd prefer the show rule for its brevity and b) this seems to be a more general shortcoming of the current Typst compiler.

**Comment by Enivex** [comment, marked as answer]
This is what you're looking for
```
#show math.equation: it =>{
  show "L" : math.cal
  it
}
```

**Reply by NicoWeio** [reply]
Nested show rules – I didn't even know that was possible. But now that I see it, it makes total sense! Thanks!

**Reply by NicoWeio** [reply]
There is one pitfall with this approach – it also applies to strings inside an equation:
```typst
#show math.equation: it =>{
  show "L" : math.cal
  it
}

$
L "is the Lagrangian."
$
```

Any idea how to handle that?

**Reply by EpicEricEE** [reply]
You could use a regex instead and only match when really only "L" is given by using start (`^`)  and end (`$`) anchors:
```typ
#show math.equation: it => {
  show regex("^L$"): math.cal
  it
}
```

In a similar fashion, you could also use `regex("\bL\b")` instead, where `\b` means "word boundary".
